### PR TITLE
Fix module loading bug in instance tracker class

### PR DESCRIPTION
**Pull request by cosmicBboy**
# TL;DR

This PR fixes an issue reported [here](https://github.com/flyteorg/flyte/issues/4245) where `pyflyte` and other executables were trying to be loaded by the `tracker.py` module when finding trackable instances.

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - ~~[ ] Smoke tested~~
 - ~~[ ] Unit tests added~~ (will be added subsequently)
 - ~~[ ] Code documentation added~~
 - ~~[ ] Any pending items have an associated Issue~~

## Complete description

The [eager workflow](https://github.com/flyteorg/flytekit/pull/1579) PR introduced a bug that caused breakages in `@dynamic` workflows and agents. The underlying reason is that eager workflows need to be able to load task/dynamic/workflow functions as `InstanceTrackingMeta` instances so that it can execute them via `FlyteRemote`. The issue is that, before `1579`, `InstanceTrackingMeta` did not support loading entities defined in the `__main__` module.

This PR handles a module loading error that was being raised because the `tracker` module was trying to load the `pyflyte` and other `pyflyte-*` executables.

## Tracking Issue

https://github.com/flyteorg/flyte/issues/4245

## Follow-up issue

https://github.com/flyteorg/flytekit/pull/1892

### CODE REVIEW

**Review by pingsutw — APPROVED** [review]
LGTM, Let's update the PR description before merging it

**Review by eapolinario — APPROVED** [review]
